### DISCUSSION TITLE
MultipleActiveResultSets

### DISCUSSION THREAD

**Discussion by neilg-dev** [oqtane/oqtane.framework#1765] [Ideas]
I very much enjoy working with Oqtane and watching how it evolves. I wondered if anyone else is building modules as I do with multiple active result sets (MARS)

Perhaps this small change could be incorporated within Oqtane.Client\Installer\Controls\SqlServerConfig.razor

		connectionString += "MultipleActiveResultSets=True;";

I tweak the code each time with this adaptation, it has no adverse impact on EF as far as I can tell.

**Comment by hishamco** [comment]
I think we can add options to use MARS or not, please push a PR if @sbwalker agree on this, if you have any issues with creating the PR I can do it in your behalf

**Comment by sbwalker** [comment]
MARS is only supported by SQL Server - so it is not a feature that could be utilized in the core framework itself (as the framework needs to be database agnostic)... however it could be used in custom modules in environments where SQL Server is the database of choice.

**Reply by hishamco** [reply]
Ya it's for SQL Server, while we support SQL Server, we could add it as a settings when SQL Server is selected

**Reply by neilg-dev** [reply]
Thank you for considering this. It's true it would only apply to custom modules with an SQL Server backend. My Oqtane implementations are all using MARS like this and the change does allow me to avoid having a bespoke framework.

**Comment by sbwalker** [comment]
@NeilGallacherSQL can you explain what you mean by "avoid having a bespoke framework" - I am not sure I understand.

**Reply by neilg-dev** [reply]
@sbwalker  At the moment I clone the repo and then make this change and then then add my modules and themes to what I consider to be a "vanilla" Oqtane. I like to keep it that way so that I can quickly update to the latest version of your dev branch (picking up for example the selector for trusted connections on the SQL config page -thank you) . I'd done my own quick workaround for that but it was great to use the update you'd released and discard my own workaround.  Hope this makes sense.  As I think I must be in a minority using MARS for modules I quite understand if the idea doesn't fit. In any case thanks again for Oqtane.
